pkg/controller/agent/reconcile.go

```diff
@@ -27,7 +27,7 @@ import (
 	"github.com/elastic/cloud-on-k8s/pkg/utils/pointer"
 )
 
-func reconcilePodVehicle(params Params, podTemplate corev1.PodTemplateSpec) *reconciler.Results {
+func reconcilePodVehicle(params Params, podTemplate corev1.PodTemplateSpec, status *agentv1alpha1.AgentStatus) *reconciler.Results {
```

Comment:
What's a bit messy here is that on the one hand we have a more functional approach where results are returned and not pointers manipulated:
```golang
reconcilePodVehicle(...) *reconcilerResults
```

and now we are introducing status here that is following another idiom where a pointer to a struct is passed and the underlying struct is manipulated: sort of a hidden return value if you will. 

```golang
reconcilePodVehicle(&status)
```

Reply:
I see your point here.  I have refactored this to return both a status and a results pointer throughout the reconciliation.  Lmk if that is what you were intending.